File: Algorithms/PrimeNumber_Algorithms/PrimeNumber_in_Range.py
#a simple algorithm that returns all the prime numbers in a specified range
from math import sqrt
base, top = map(int, input("Range:").split())

def isPrime(z):
    isprime = True
    if z <= 1:
        isprime = False
    else:
        for i in range(2, int(sqrt(z)) + 1):
            if z % i == 0:
                isprime = False
                break
    return isprime

for i in range(base, top + 1):
    if isPrime(i) == True:
        print(i)

File: Algorithms/PrimeNumber_Algorithms/PrimeNumber_in_Range.cpp
//a simple algorithm that returns all the prime numbers in a specified range
#include <iostream>
#include <math.h>
using namespace std;

bool isPrime(int n) {
  bool isPrime = true;
  if (n <= 1) {
    isPrime = false;
  }
  else {
    for (int i = 2; i <= sqrt(n); i++) {
      if (n % i == 0) {
        isPrime = false;
        break;
      }
    }
  }
  return isPrime;
}

int main() {
    int base, top;
    cout << "Range:";
    cin >> base >> top;

    for (int i = base; i <= top; i++) {
      if (isPrime(i) == true) {
        cout << i << " ";
      }
    }

    cout << endl;

    return 0;
}